qiskit/circuit/library/n_local/n_local.py

```diff
@@ -1069,3 +1299,170 @@ def _stdlib_gate_from_simple_block(block: QuantumCircuit) -> _StdlibGateResult |
     ):
         return None
     return _StdlibGateResult(instruction.operation.base_class, len(instruction.operation.params))
+
+
+def _normalize_entanglement(
+    entanglement: (
+        BlockEntanglement
+        | Iterable[BlockEntanglement]
+        | Callable[[int], BlockEntanglement | Iterable[BlockEntanglement]]
+    ),
+    num_entanglement_blocks: int,
+) -> list[str | list[tuple[int]]] | Callable[[int], list[str | list[tuple[int]]]]:
+    """If the entanglement is Iterable[Iterable], normalize to list[tuple]."""
+    if isinstance(entanglement, str):
+        return [entanglement] * num_entanglement_blocks
+
+    elif isinstance(entanglement, Iterable):
```

Comment:
So checking `isinstance` `Iterable` isn't really the ideal way to check if something is actually iterable. It only checks if the class has been registered as an Iterable using the abc mechanisms (either as a subclass or manual registration). But python will let you iterate over some objects if `__getitem__` is implemented for sequences. The only surefire way to detect if something is actually iterable is to try running `iter()`.

Reply:
We can just turn this around and use `not callable(entanglement)` as check, then the only valid type would be the `Iterable` 🙂